lib/krikri/ldp/invalidatable.rb

```diff
@@ -0,0 +1,63 @@
+module Krikri::LDP
+  ##
+  # Implements invalidatation for `Krikri::LDP::Resource`s. This is different 
+  # from deletetion, in that the resource continues to respond `200 OK`, and 
+  # return the representation, Nothing is removed from the LDP server.
+  module Invalidatable
+    INVALIDATED_BY_URI = RDF::PROV.wasInvalidatedBy
+    INVALIDATED_TIME_URI = RDF::PROV.invalidatedAtTime
+
+    ##
+    # Invalidates the resource by marking it with a `prov:invalidatedAtTime`.
+    #
+    # @return [void]
+    def invalidate!(activity_uri = nil)
+      raise "Cannot invalidate #{rdf_subject}, does not exist." unless exists?
+
+      # force a reload unless we have cached an invalidatedAtTime
+      rdf_source.get({}, true) unless invalidated?
+      # we check invalidated again in case the reload came back invalid
+      raise "Cannot invalidate #{rdf_subject}, already invalid." if invalidated?
+
+      uri = RDF::URI(rdf_subject)
+      
+      rdf_source << [uri, INVALIDATED_BY_URI, activity_uri] unless 
+        activity_uri.nil?
+      rdf_source << [uri, INVALIDATED_TIME_URI, DateTime.now]
+
+      rdf_source.save
+    end
+
+    ##
+    # @return [Boolean] `true` if the resource has been marked invalidated.
+    def invalidated?
+      !invalidated_at_time.nil?
```

Comment:
FWIW, PROV-DM [seems to suggest](http://www.w3.org/TR/2013/REC-prov-dm-20130430/#term-Invalidation) that since Invalidation of an Entity occurs within the context of an Activity, it should probably be `prov:wasInvalidatedBy`.